src/Utils/actionCondition.ts

```diff
@@ -0,0 +1,123 @@
+}
+
+export const getEnumConditionName = (entity: CLM.EntityBase, enumValue: CLM.EnumValue): string => {
+    return `${entity.entityName} == ${enumValue.enumValue}`
+}
+
+export const getValueConditionName = (entity: CLM.EntityBase, condition: CLM.Condition): string => {
+    return `${entity.entityName} ${conditionDisplay[condition.condition]} ${condition.value}`
+}
+
+export const convertConditionToConditionalTag = (condition: CLM.Condition, entities: CLM.EntityBase[]): IConditionalTag => {
+    const entity = entities.find(e => e.entityId === condition.entityId)
+    if (!entity) {
+        throw new Error(`Condition refers to non-existent Entity ${condition.entityId}`)
+    }
+
+    let conditionalTag: IConditionalTag
+    if (entity.entityType === CLM.EntityType.ENUM) {
+        if (!entity.enumValues) {
+            throw new Error(`Condition refers to Entity without Enums ${entity.entityName}`)
+        }
+
+        const enumValueId = condition.valueId
+        if (!enumValueId) {
+            throw new Error(`Condition refers to enum entity, but condition did not have enum value id.`)
+        }
+        const enumValue = entity.enumValues.find(e => e.enumValueId === enumValueId)
+        if (!enumValue) {
+            throw new Error(`Condition refers to non-existent EnumValue: ${enumValueId}`)
+        }
+
+        const name = getEnumConditionName(entity, enumValue)
+        conditionalTag = {
+            key: enumValue.enumValueId!,
+            name,
+            condition,
+        }
+    }
+    else {
+        const name = getValueConditionName(entity, condition)
+        const key = CLM.hashText(name)
+        conditionalTag = {
+            key,
+            name,
+            condition,
+        }
+    }
+
+    return conditionalTag
+}
+
+export const isConditionEqual = (conditionA: CLM.Condition, conditionB: CLM.Condition): boolean => {
+    return conditionA.entityId === conditionB.entityId
+        && conditionA.condition === conditionB.condition
+        && conditionA.valueId === conditionB.valueId
+        && conditionA.value === conditionB.value
+}
+
+
+/**
+ * Given memory value,
+ * If entity is multivalue returns the number of labels/values
+ * If entity has number resolution return number from label
+ * Otherwise, return undefined to indicate number can't be parsed from memory
+ */
+export const findNumberFromMemory = (memory: CLM.Memory, isMultivalue: boolean): number | undefined => {
+    if (isMultivalue) {
+        return memory.entityValues.length
+    }
+
+    const valueString: string | undefined = memory
+        && memory.entityValues[0]
+        && (memory.entityValues[0].resolution ? true : undefined)
+        && (memory.entityValues[0].resolution as any).value as string
+
+    const value = valueString
+        ? parseInt(valueString)
+        : undefined
+
+    return value
+}
+
+export const isValueConditionTrue = (condition: CLM.Condition, numberValue: number): boolean => {
```

Comment:
This method feels like it belongs in Models, where the conditions are applied.

Reply:
Saw your other PR on models repo and reminded me of this comment. I would say Models was named that way because it was originally intended to only be models. As a set of types/interfaces shared by the UI and SDK to ensure they transferred same objects over HTTP.  Since types and interface don't translate to any material code this made the repo extremely light weight. There has been more and more stuff put in there and it's starting to become something other than models, but in general it was only moved if it was required to be the same and actually used in both places.

That being said, there is a different `isValueConditionTrue` in SDK that might be appropriate to consolidate.

Reply:
SDK also SGTM